lib/simp/rake/build/pkg.rb

```diff
@@ -728,63 +638,168 @@ def define_tasks
+
+            require_rebuild?(dir, { :verbose => true, :check_git => true })
+          end
+        end
+
         ##############################################################################
         # Helper methods
         ##############################################################################
 
-        # Takes a list of directories to hop into and perform builds within
-        # Needs to be passed the chroot path as well
+        # Generate a random string suitable for a rake task namespace
         #
-        # The task must be passed so that we can output the calling name in the
-        # status bar.
-        def build(chroot,dirs,task,add_to_autoreq=true,snapshot_release=false)
-          validate_in_mock_group?
-          _verbose = ENV.fetch('SIMP_PKG_verbose','no') == 'yes'
+        # This is used as a workaround for Parallelization
+        def generate_namespace
+          return (0...24).map{ (65 + rand(26)).chr }.join.downcase
+        end
+
+        # Check and see if 'dir' requires a rebuild based on published packages
+        #
+        # If 'fetch' is true => Download the RPM
+        # If 'verbose' is true => Print helpful information to stderr
+        # If 'check_git' is true => Print the git tag status if 'verbose' is true
+        def require_rebuild?(dir, opts={ :unique_namespace => generate_namespace, :fetch => false, :verbose => @verbose, :check_git => false })
+          result = false
 
-          rpm_dependency_file = File.join(@base_dir, 'build', 'rpm',
-            'dependencies.yaml')
+          rpm_dependency_file = File.join(@base_dir, 'build', 'rpm', 'dependencies.yaml')
 
           if File.exist?(rpm_dependency_file)
             rpm_metadata = YAML.load(File.read(rpm_dependency_file))
           end
 
-          metadata = Parallel.map(
+          Dir.chdir(dir) do
+            if File.exist?('metadata.json')
+              # Generate RPM metadata files
+              # - 'build/rpm_metadata/requires' file containing RPM
+              #   dependency/obsoletes information from the
+              #   'dependencies.yaml' and the module's
+              #   'metadata.json'; always created
+              # - 'build/rpm_metadata/release' file containing RPM
+              #   release qualifier from the 'dependencies.yaml';
+              #   only created if release qualifier if specified in
+              #   the 'dependencies.yaml'
+              Simp::Rake::Build::RpmDeps::generate_rpm_meta_files(dir, rpm_metadata) if rpm_metadata
+            end
+
+            new_rpm = Simp::Rake::Pkg.new(Dir.pwd, opts[:unique_namespace], @simp_version)
+
+            new_rpm_info = Simp::RPM.new(new_rpm.spec_file)
+
+            if opts[:check_git]
+              require_tag = false
+
+              # We remove any leading 'v' chars since some projects use them
+              latest_tag = %x(git describe --abbrev=0 --tags 2>/dev/null).strip.gsub(/^v/,'')
```

Comment:
This does not work for our forked modules that have tags that begin with simp-, simp6.0.0-, or have suffixes.  So, these modules get erroneously rebuilt.